src/tests/unit/platform/test_policies.py

```diff
@@ -0,0 +1,522 @@
+# * NON-INFRINGEMENT AND FITNESS FOR A PARTICULAR PURPOSE.
+
+"""Tests of the policies support in the Platform API."""
+import copy
+
+import pytest
+import logging
+import random
+from contextlib import ExitStack as does_not_raise
+from cbc_sdk.rest_api import CBCloudAPI
+from cbc_sdk.platform import Policy, PolicyRule
+from cbc_sdk.errors import ApiError, InvalidObjectError, ServerError
+from tests.unit.fixtures.CBCSDKMock import CBCSDKMock
+from tests.unit.fixtures.platform.mock_policies import (FULL_POLICY_1, SUMMARY_POLICY_1, SUMMARY_POLICY_2,
+                                                        SUMMARY_POLICY_3, OLD_POLICY_1, FULL_POLICY_2, OLD_POLICY_2,
+                                                        RULE_ADD_1, RULE_ADD_2, RULE_MODIFY_1, NEW_POLICY_CONSTRUCT_1,
+                                                        NEW_POLICY_RETURN_1)
+
+
+logging.basicConfig(format='%(asctime)s %(levelname)s:%(message)s', level=logging.DEBUG, filename='log.txt')
+
+
+@pytest.fixture(scope="function")
+def cb():
+    """Create CBCloudAPI singleton"""
+    return CBCloudAPI(url="https://example.com",
+                      org_key="test",
+                      token="abcd/1234",
+                      ssl_verify=False)
+
+
+@pytest.fixture(scope="function")
+def cbcsdk_mock(monkeypatch, cb):
+    """Mocks CBC SDK for unit tests"""
+    return CBCSDKMock(monkeypatch, cb)
+
+
+# ==================================== UNIT TESTS BELOW ====================================
+
+def test_policy_compatibility_aliases_read(cb):
+    """Test the compatibility aliases that mimic the behavior of the old policy object."""
+    policy = Policy(cb, 65536, FULL_POLICY_1, False, True)
+    assert policy.priorityLevel == "HIGH"
+    assert policy.systemPolicy is False
+    assert policy.version == 2
+    assert policy.latestRevision == 2
+    assert policy.policy == OLD_POLICY_1
+    assert policy.rules == FULL_POLICY_1["rules"]
+    objs = policy.object_rules
+    for raw_rule in FULL_POLICY_1["rules"]:
+        assert objs[raw_rule["id"]]._info == raw_rule
+
+
+def test_policy_compatibility_aliases_write(cb):
+    """Test the compatibility aliases that mimic the behavior of the old policy object."""
+    policy = cb.create(Policy)
+    policy.policy = copy.deepcopy(OLD_POLICY_2)
+    policy.description = "Hoopy Frood"
+    policy.name = "default - S1"
+    policy.priorityLevel = "MEDIUM"
+    policy.version = 2
+    new_policy_data = copy.deepcopy(policy._info)
+    assert new_policy_data == FULL_POLICY_2
+
+
+def test_policy_autoload(cbcsdk_mock):
+    """Tests automatic loading of a "partial" policy."""
+    called_full_get = False
+
+    def on_get(uri, query_params, default):
+        nonlocal called_full_get
+        called_full_get = True
+        return FULL_POLICY_1
+
+    cbcsdk_mock.mock_request('GET', '/policyservice/v1/orgs/test/policies/65536', on_get)
+    api = cbcsdk_mock.api
+    policy = Policy(api, 65536, SUMMARY_POLICY_1, False, False)
```

Comment:
Shouldn't we only need to provide the following to get the policy using the class directly?

```
Policy(api, 65536)

or

cb.select(Policy, 65536)
```

Reply:
This isn't looking up a policy, it's creating one locally for use as part of the test.

Reply:
It's creating a policy object with that id in order to fetch the content of that from the Carbon Black Cloud otherwise this would need a POST instead of GET. I believe the above code is equivalent to what you have there currently.

Reply:
Line 88 does not "get" a policy. It is _simulating_ a `Policy` object initialized only from summary information, as, for example, would be done by `PolicyQuery`.  The objective here is to show that reading a property that was already initialized from summary information does _not_ trigger a full GET, but reading a property that was _not_ so initialized will trigger a full GET.